pkg/plugins/file/main.go

```diff
@@ -1,9 +1,82 @@
 package file
 
-// File defines the resource file that can be configured
-// from an updatecli configuration
+import (
+	"fmt"
+	"strings"
+
+	"github.com/updatecli/updatecli/pkg/core/text"
+)
+
+// FileSpec defines a specification for a "file" resource
+// parsed from an updatecli manifest file
+type FileSpec struct {
+	File           string
+	Line           int
+	Content        string
+	ForceCreate    bool
+	MatchPattern   string
+	ReplacePattern string
+}
+
+// File defines a resource of type "file"
 type File struct {
-	File    string
-	Line    Line
-	Content string
+	spec             FileSpec
+	contentRetriever text.TextRetriever
+	CurrentContent   string
+}
+
+// New returns a reference to a newly initialized File object from a Filespec
+// or an error if the provided Filespec triggers a validation error.
+func New(spec FileSpec) (*File, error) {
```

Comment:
I think it would be better to run all validation, collect every error in a list of errors and then exit an error if there is at least one error. The purpose is to detect every potential configuration issue.

Reply:
It had been applied as we discussed: The method has a list of error strings, and they are returned one time for all.